src/test/recovery/t/050_check_recovery_backup.pl

```diff
@@ -0,0 +1,162 @@
+		"'$label': rm primary pgdata");
+	$primary->command_ok([ 'cp', '-r', $backup_dir, $primary_pgdata ],
+		"'$label': copy basebackup to primary");
+
+	# configure recovery
+	$primary->set_recovery_mode;
+	open my $conf, '>', "$primary_pgdata/postgresql.auto.conf";
+	print $conf "recovery_target_action = 'promote'\n";
+	print $conf "restore_command = 'cp -vr $archivedir/%f $primary_pgdata/%p'\n";
+	print $conf "$restore_options\n";
+	close $conf;
+	$primary->start;
+}
+
+sub wait_recovery_and_switch_wal_primary
+{
+	my ($label) = @_;
+
+	print("Running: wait_recovery_and_switch_wal_primary: '$label'\n");
+
+	$primary->poll_query_until('test_db', "SELECT true FROM pg_switch_wal()")
+		or die "'$label': Timed out while waiting for switch WAL after recovery";
+}
+
+sub restore_backup_standby
+{
+	my ($backup_dir, $restore_point, $primary_host, $primary_port, $label) = @_;
+
+	my $standby_pgdata = $standby->data_dir;
+	my $standby_port = $standby->port;
+
+	$standby->command_ok([ 'rm', '-rf', $standby_pgdata ],
+		"'$label': rm standby pgdata");
+	$standby->command_ok([ 'cp', '-r', $backup_dir, $standby_pgdata ],
+		"'$label': copy basebackup to standby");
+
+	$standby->set_recovery_mode;
+	open my $conf, '>', "$standby_pgdata/postgresql.auto.conf";
+	print $conf "synchronous_standby_names = '*'\n";
+	print $conf "recovery_target_action = 'shutdown'\n";
+	print $conf "restore_command = 'cp -vr $archivedir/%f $standby_pgdata/%p'\n";
+	print $conf "recovery_target_name = '$restore_point'\n";
+	close $conf;
+	$standby->adjust_conf('postgresql.conf', 'port', $standby_port);
+	$standby->start(fail_ok => 1);
+
+	$standby->wait_for_log('database system is shut down');
+
+	# configure primary connection with replication slot
+	unlink "$standby_pgdata/recovery.signal";
+	unlink "$standby_pgdata/postgresql.auto.conf";
+	$standby->set_standby_mode;
+	open $conf, '>', "$standby_pgdata/postgresql.auto.conf";
+	print $conf "primary_conninfo = ";
+	print $conf "'user=postgres port=" . $primary_port;
+	print $conf " host=" . $primary_host;
+	print $conf " sslmode=prefer sslcompression=1 gssencmode=prefer";
+	print $conf " krbsrvname=postgres target_session_attrs=any";
+	print $conf " application_name=gp_walreceiver'\n";
+	print $conf "primary_slot_name = 'internal_wal_replication_slot'\n";
+	close $conf;
+
+	$standby->_update_pid(0);
+	$standby->start;
+}
+
+mkdir($archivedir);
+
+# Initialize primary
+$primary->init(allows_streaming => 1, extra => ['--data-checksums']);
+$primary->append_conf("postgresql.conf", "archive_mode = 'on'");
+$primary->append_conf("postgresql.conf", "archive_command = 'cp -rv %p $archivedir/'");
+$primary->start;
+$primary->safe_psql('postgres', 'CREATE DATABASE test_db');
+$primary->safe_psql('test_db', 'CREATE TABLE test AS SELECT generate_series(1,10)');
+$primary->safe_psql('postgres', 'CREATE ROLE postgres WITH LOGIN replicaTION');
+
+# Initialize standby
+$standby->init(extra => ['--data-checksums']);
```

Comment:
What is the point of this command? And in general, `standby` is usually not initialized like this, but with `init_from_backup`.